controllers/utils.go

```diff
@@ -470,3 +471,46 @@ func addRestoreLabelSelector(
 		)
 	}
 }
+
+// update BackupSchedule status and remove velero schedules
+// when the BackupSchedule is paused or in BackupCollision
+func updateBackupSchedulePhaseWhenPaused(
+	ctx context.Context,
+	c client.Client,
+	veleroScheduleList veleroapi.ScheduleList,
+	backupSchedule *v1beta1.BackupSchedule,
+	phase v1beta1.SchedulePhase,
+	msg string,
+) error {
+	scheduleLogger := log.FromContext(ctx)
+
+	if phase != v1beta1.SchedulePhasePaused && phase != v1beta1.SchedulePhaseBackupCollision {
+		// do not process any other type of schedule here
+		return nil
+	}
+
+	//update status
+	backupSchedule.Status.Phase = phase
+	backupSchedule.Status.LastMessage = msg
+
+	backupSchedule.Status.VeleroScheduleCredentials = nil
+	backupSchedule.Status.VeleroScheduleManagedClusters = nil
+	backupSchedule.Status.VeleroScheduleResources = nil
+
+	err := c.Status().Update(ctx, backupSchedule)
+
+	// delete schedules, so we don't generate new backups
+	for i := range veleroScheduleList.Items {
+		scheduleLogger.Info(
+			"Attempt to delete schedule " + veleroScheduleList.Items[i].Name,
+		)
+
+		if err := c.Delete(ctx, &veleroScheduleList.Items[i]); err == nil {
```

Comment:
I think we should check for err != nil here and return the error if it occurs - otherwise if there's an error deleting a veleroschedule we may not retry until the next reconcile (whenever that happens)

Reply:
I am checking if the schedule throws an error, and ignore the resource not found error. If any other delete errors, I call a reconcile and not update the BackupSchedule status